src/renderer/redux/actions/subscriptions.js

```diff
@@ -14,5 +21,98 @@ export const doChannelUnsubscribe = (subscription: Subscription) => (dispatch: D
     data: subscription,
   });
 
+export const doCheckSubscriptions = () => (
+  dispatch: Dispatch,
+  getState: () => SubscriptionState
+) => {
+  const checkSubscriptionsTimer = setInterval(
+    () =>
+      selectSubscriptions(getState()).map((subscription: Subscription) =>
+        dispatch(doCheckSubscription(subscription))
+      ),
+    CHECK_SUBSCRIPTIONS_INTERVAL
+  );
+  dispatch({
+    type: ACTIONS.CHECK_SUBSCRIPTIONS_SUBSCRIBE,
+    data: { checkSubscriptionsTimer },
+  });
+};
+
+export const doCheckSubscription = (subscription: Subscription) => (dispatch: Dispatch) => {
+  dispatch({
+    type: ACTIONS.CHECK_SUBSCRIPTION_STARTED,
+    data: subscription,
+  });
+
+  Lbry.claim_list_by_channel({ uri: subscription.uri, page: 1 }).then(result => {
+    const claimResult = result[subscription.uri] || {};
+    const { claims_in_channel: claimsInChannel } = claimResult;
+
+    let count = subscription.latest
+      ? claimsInChannel.reduce(
+          (prev, cur, index) =>
+            buildURI({ contentName: cur.name, claimId: cur.claim_id}, false) === subscription.latest ? index : prev,
+          -1
+        )
+      : 1;
+
+    if (count !== 0) {
+      if (!claimsInChannel[0].value.stream.metadata.fee) {
+        dispatch(
+          doPurchaseUri(buildURI({ contentName: claimsInChannel[0].name, claimId: claimsInChannel[0].claim_id }, false), { cost: 0 })
```

Comment:
Why do you need to pass in the new cost object. Shouldn't `doPurchaseUri` be able to select that data with the uri?

Reply:
its is more efficient this way because I already have the cost information so we don't need to fetch it again and